src/routes/safe/components/Balances/SendModal/screens/ContractInteraction/utils/index.ts

```diff
@@ -73,7 +74,20 @@ export const formMutators = {
   },
 }
 
-export const handleSubmitError = (error, values) => {
+export const isAddress = (type: string): boolean => type.indexOf('address') === 0
+export const isBoolean = (type: string): boolean => type.indexOf('bool') === 0
+export const isString = (type: string): boolean => type.indexOf('string') === 0
+export const isUint = (type: string): boolean => type.indexOf('uint') === 0
+export const isInt = (type: string): boolean => type.indexOf('int') === 0
+export const isByte = (type: string): boolean => type.indexOf('byte') === 0
+
+export const isNotStringType = (item: string): boolean => typeof item !== 'string'
+export const isNotBooleanType = (item: string): boolean => typeof item !== 'boolean'
+export const isNotNumberType = (item: string): boolean => !Number.isInteger(item)
+
+export const isArrayParameter = (parameter: string): boolean => /\[]$/.test(parameter)
```

Comment:
This check is not good enough, there are arrays of fixed length in solidity where there is an additional number between the brackets. See https://solidity.readthedocs.io/en/latest/abi-spec.html

Reply:
There are also tuples, would you add this in a different pr?

Reply:
Oh! Thanks for the insight. Yes, I'll work on tuples after this PR is merged.

To be sure I'm not missing something:
- `bytes` and `strings` are dynamic-sized but not represented as an array. No required action.
- `T[]`, is what this PR fixed
- `T[k]`. Not sure I'm understanding the `k >= 0`, how is the case for `k == 0`? What should we expect?

Reply:
you do not need to check the length (this is done by the web3js encoder), just in case of `T[k]` it should expect an array (no matter what k is)

Also be aware that you can have `T[][][4][5]` or any other combination. As your current endpoint just checks that it ends on `[]` this should alreay work for non fixed arrays, so for this PR only fixed arrays need to be handled.

Reply:
fix for arrays: [e9c20a8](https://github.com/gnosis/safe-react/pull/1009/commits/e9c20a8e5c66da01ea582dff37e22a53f8bf7695)

The issue for the `tuple` type: #1028